presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/PasswordValidator.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.file;
+
+import com.google.common.cache.Cache;
+import io.airlift.log.Logger;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import static io.prestosql.plugin.password.file.EncryptionUtil.checkBCryptPassword;
+import static io.prestosql.plugin.password.file.EncryptionUtil.checkPBKDF2Password;
+
+class PasswordValidator
+{
+    private static final Logger log = Logger.get(PasswordValidator.class);
+
+    private PasswordValidator() {}
+
+    public static boolean authenticate(String inputUsername, String inputPassword, String storedUsername, String storedPassword, Cache<PasswordStore, Boolean> authCacheStore, Optional<HashingAlgorithm> hashingAlgo)
+    {
+        PasswordStore passSet = new PasswordStore(inputUsername, inputPassword);
+        if (authCacheStore != null) {
+            try {
+                return authCacheStore.get(passSet, new CustomCacheLoaderCallable(passSet, storedUsername, storedPassword, hashingAlgo));
+            }
+            catch (ExecutionException e) {
+                log.error(e, "Error reading from the Authentication Cache: ");
```

Comment:
Instead of logging, let's propagate this as a runtime exception so that it's handled by the server as an error rather than an authentication failure. This matches what we do in the LDAP authenticator.
```java
throw new RuntimeException(e);

Reply:
Done.